### PR TITLE
add Lucene as dependency to the web module

### PR DESCRIPTION
**Pull request by vladak**
IDEA was unhappy because it could not resolve dependency to compile `SuggesterQueryBuilder.java` due to the imports:
```
import org.apache.lucene.queryparser.classic.ParseException;
import org.apache.lucene.search.Query;
```

This change makes it happy again.

### CODE REVIEW

File: opengrok-web/pom.xml [hunk 1]
```diff
@@ -101,6 +101,11 @@
             <artifactId>org.suigeneris.jrcs.diff</artifactId>
             <version>0.4.2</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.lucene</groupId>
+            <artifactId>lucene-queryparser</artifactId>
+            <version>7.3.1</version>
```
Comment (Orviss) [inline opengrok-web/pom.xml]: Please use Lucene version variable.

**Comment by Orviss** [comment]
Is this really necessary? It works for me without problems.

**Comment by Orviss** [comment]
If you look at `indexer` or `suggester`, both of these have this dependency. And since `web` depends on them, then it has it transitively. What exact error does IDEA show?

**Comment by vladak** [comment]
It underlines the imports with red.

**Comment by Orviss** [comment]
> It underlines the imports with red.

This does not happen to me. I will try clean checkout later this evening.

**Comment by Orviss** [comment]
Clean checkout. Imported from external model (Maven). IDEA 2018.1 Ultimate.

Imports are not underlined.

screen shot 2018-08-16 at 20 05 49

**Comment by vladak** [comment]
Maybe this has to do with my use case - I am opening each Maven project in separate IDEA window. Will re-check.

**Comment by tarzanek** [comment]
seen the same here when java EE module with tomcat support wasn't enabled
resp. when I imported project BEFORE I enabled the tomcat support in Idea Ultimate
I suppose Idea CE will have the same prob ... 
(and I am wondering about Netbeans, maybe they work out of box)

**Comment by ahornace** [comment]
`lucene` should be added transitively. Please reopen if needed.

**Pull request closed without merging**